### PR TITLE
Add hint about inline link issue

### PR DESCRIPTION
**Pull request by malteneuss**
Quickfix for https://github.com/haskell/haddock/issues/774. Feel free to edit the phrasing.

### CODE REVIEW

File: doc/markup.rst [hunk 1]
```diff
@@ -1121,6 +1121,17 @@ followed by the URL enclosed in regular parentheses, for example: ::
 The link text is used as a description for the URL if the output
 format supports it.
 
+Hint: There's a `known issue <https://github.com/haskell/haddock/issues/774>`_ 
+that any inline link at the beginning of a line within a multi-line comment 
+isn't rendered correctly: ::
+   
+    {-| Some multi-line comment that has a
+    [link](https://not-rendering.org) and a
+    [reference link]: https://not-rendering.org
```
Comment (Kleidukos) [inline doc/markup.rst:1130]: The domain `not-rendering.org` should not be used, it risks being resolved and queries by automated tools, and bring unwanted traffic to the website, even if it is not registered right now.

Prefer `example.com` instead.

**Review by Kleidukos — CHANGES_REQUESTED** [review]
Hi, thank you for the PR. I've left a change request and it should be good!

File: doc/markup.rst (hunk 1 above)
Comment (malteneuss) [reply doc/markup.rst]: Fixed. Thanks for the info.

**Review by Kleidukos — APPROVED** [review]
LGTM!

**Pull request merged**